pkg/virt-handler/vm_test.go

```diff
@@ -1462,6 +1470,72 @@ var _ = Describe("VirtualMachineInstance", func() {
 		})
 
 	})
+	Context("When VirtualMachineInstance is connected to a network", func() {
+
+		It("should only report the pod network in status", func() {
+			vmi := v1.NewMinimalVMI("testvmi")
+			vmi.UID = vmiTestUUID
+			vmi.ObjectMeta.ResourceVersion = "1"
+			vmi.Status.Phase = v1.Scheduled
+			vmi.Status.Interfaces = make([]v1.VirtualMachineInstanceNetworkInterface, 0)
+			vmi = addActivePods(vmi, podTestUUID, host)
+
+			podCacheInterface := network.PodCacheInterface{
+				Iface: &v1.Interface{
+					Name: "default",
+				},
+				PodIP: "1.1.1.1",
+			}
+			podJson, err := json.Marshal(podCacheInterface)
+			Expect(err).ToNot(HaveOccurred())
+			err = os.MkdirAll(fmt.Sprintf(util.VMIInterfaceDir, vmi.UID), 0755)
+			Expect(err).ToNot(HaveOccurred())
+			vmiInterfacepath := fmt.Sprintf(util.VMIInterfacepath, vmi.UID, "default")
+			f, err := os.Create(vmiInterfacepath)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = f.WriteString(string(podJson))
+			Expect(err).ToNot(HaveOccurred())
+			f.Close()
+
+			updatedVMI := vmi.DeepCopy()
+			updatedVMI.Status.Phase = v1.Running
+			updatedVMI.Status.Conditions = []v1.VirtualMachineInstanceCondition{
+				{
+					Type:   v1.VirtualMachineInstanceIsMigratable,
+					Status: k8sv1.ConditionTrue,
+				},
+			}
+			updatedVMI.Status.MigrationMethod = v1.LiveMigration
+			mockWatchdog.CreateFile(vmi)
+			domain := api.NewMinimalDomainWithUUID("testvmi", vmiTestUUID)
+			domain.Status.Status = api.Running
+			vmiFeeder.Add(vmi)
+			domainFeeder.Add(domain)
+
+			node := &k8sv1.Node{
+				Status: k8sv1.NodeStatus{
+					Addresses: []k8sv1.NodeAddress{
+						{
+							Type:    k8sv1.NodeInternalIP,
+							Address: "127.0.0.1",
+						},
+					},
+				},
+			}
+			fakeClient := fake.NewSimpleClientset(node).CoreV1()
+			virtClient.EXPECT().CoreV1().Return(fakeClient).AnyTimes()
+
+			vmiInterface.EXPECT().Update(NewVMICondMatcher(*updatedVMI))
+
+			controller.Execute()
+
+			vmiInterface.EXPECT().Update(gomock.Any()).Do(func(arg interface{}) {
```

Comment:
This will fail because it's being called after controller.Execute... basically there's no way this Update will ever occur because there's no logic to call it once the expect is defined.

so, 

- define expect
- execute controller

But here we have

- execute controller
- define expect

Reply:
Corrected now